### PR TITLE
Implement `Bits` and `BMap`

### PR DESCRIPTION
**Pull request by edusporto**

### CODE REVIEW

File: book/BMap/has.kind2
Comment (developedby) [inline book/BMap/has.kind2:1]: All these functions to get a property should by default not consume the map, but instead return a pair like the `got` and be linear (not duplicate anything unnecessary).
Having the option of a function that consumes the map but has a simpler return is ok, but realistically it won't be used as often

File: book/BMap/size.kind2
Comment (developedby) [inline book/BMap/size.kind2:1]: Same here

File: book/BMap/set.kind2 [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+// Replaces the old value if the given path is already in the map.
+//
+// Bits/o traverses to the left and Bits/i to the right.
+set <V> (path: Bits) (val: (Maybe V)) (map: (BMap V)) : (BMap V) =
+  match map {
+    leaf: match path {
+      e: (node V (leaf V) val (leaf V))
+      // NOTE: New intermediary notes have their value set to `none`
+      o: (node V (set path.pred val (leaf V)) none (leaf V))
+      // NOTE: New intermediary notes have their value set to `none`
```
Comment (developedby) [inline book/BMap/set.kind2:15]: Don't need this note twice

**Review by developedby — CHANGES_REQUESTED** [review]

**Comment by edusporto** [comment]
@developedby suggested changes resolved

**Pull request closed without merging**